source/grid/manifold.cc

```diff
@@ -87,6 +87,68 @@ get_new_point (const Quadrature<spacedim> &quad) const
   return p;
 }
 
+template <int dim, int spacedim>
+Point<spacedim>
+Manifold<dim, spacedim>::
+get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
+               const std::vector<double>           &weights) const
+{
+  const double tol = 1e-10;
+  const unsigned int n_points = surrounding_points.size();
+
+  Assert(n_points > 0,
+         ExcMessage("There should be at least one point."));
+
+  Assert(n_points == weights.size(),
+         ExcMessage("There should be as many surrounding points as weights given."));
+
+  Assert(std::abs(std::accumulate(weights.begin(), weights.end(), 0.0)-1.0) < tol,
+         ExcMessage("The weights for the individual points should sum to 1!"));
+
+  // The number of points is small (at most 26 in 3D), therefore try to minize
+  // overhead and use a very simple search of O(N^2) to find the order in which
+  // to loop over points.
+  std::vector<unsigned int> permutation(n_points);
+  std::vector<bool> found(n_points,false);
```

Comment:
Hmm, I really do not see where this is happening. I have looked through all calling functions, and can not find the place you are referring to. There are places in `MappingManifold` that compute as many new points as necessary for the mapping degree, but it does so by always using the same cell vertices and changing the weights of them for each new quadrature point. And `MappingQ::get_intermediate_points` calls this function with a vector of points, but this vector can only have 2, 4 or 8 points, otherwise it throws an exception. Then there are a number of calling places in `tria_accessor.cc`, but they all call the `get_new_point_on_...` function variants that use the default points and weights (maximum 26 points). Finally there are a lot of calls from `tria_boundary_lib.cc`. Those are pretty confusing, but all of them end up calling `FlatManifold::get_new_point` instead of `Manifold:get_new_point` so they never see this algorithm.
 Also, as a test I have added output and asserts to make sure the number of points never exceeds 26, but still all mapping and manifold tests succeed. I also explicitly made a test with a higher order mapping, and a SphericalManifold but still no call with more than 26 points.
But what about the following: Take a look at #3066. Would that be ok with you?